src/HeapedVectors.jl

```diff
@@ -0,0 +1,150 @@
+__precompile__()
+
+struct HeapedVector{T, F<:Function}  
+    data::Vector{T}
+    by::F
+    function HeapedVector(v::Vector{T}, by::F) where {T, F}
+        new{T, F}(heaping(v, by), by)
+    end
+end
+      
+HeapedVector(data::Vector{T}) where {T} = HeapedVector(data, identity)  
+
+       
+function heaping(v, by)  
+    ar = typeof(v[1])[]
+    for i = 1:length(v) 
+        insert!(ar, i, v[i])
+        floatup(length(ar), ar, by)
+    end
+    return ar
+end
+
+function floatup(index, ar, by)	
```

Comment:
Should be `floatup!` and `ar` should be the first argument, since it is modified.

Reply:
ok I'll chaange it to `floatup!` and change the order of argument.